pkg/gadgets/trace/tcpconnect/tracer/bpf/tcpconnect.bpf.c

```diff
@@ -93,7 +117,22 @@ enter_tcp_connect(struct pt_regs *ctx, struct sock *sk)
 	if (filter_uid != (uid_t) -1 && uid != filter_uid)
 		return 0;
 
-	bpf_map_update_elem(&sockets, &tid, &sk, 0);
+	if (calculate_latency) {
+		task = (struct task_struct*)bpf_get_current_task();
+		mntns_id = (u64) BPF_CORE_READ(task, nsproxy, mnt_ns, ns.inum);
+
+		if (filter_by_mnt_ns && !bpf_map_lookup_elem(&mount_ns_filter, &mntns_id))
+			return 0;
```

Comment:
Note: to refactor in case #1580 gets merged before.

Reply:
I'm investigating a regression on that one, so this should go in first.